stubs/pluggy/pluggy/_hooks.pyi

```diff
@@ -0,0 +1,117 @@
+from collections.abc import Callable, Generator, Mapping, Sequence, Set as AbstractSet
+from types import ModuleType
+from typing import Any, TypeVar, overload
+from typing_extensions import Final, TypeAlias, TypedDict
+
+from ._result import _Result
+
+_T = TypeVar("_T")
+_F = TypeVar("_F", bound=Callable[..., object])
+_Namespace: TypeAlias = ModuleType | type
+_Plugin = object
+_HookExec: TypeAlias = Callable[[str, Sequence[HookImpl], Mapping[str, object], bool], object | list[object]]
+_HookImplFunction: TypeAlias = Callable[..., _T | Generator[None, _Result[_T], None]]
+
+class _HookSpecOpts(TypedDict):
+    firstresult: bool
+    historic: bool
+    warn_on_impl: Warning | None
+
+class _HookImplOpts(TypedDict):
+    wrapper: bool
+    hookwrapper: bool
+    optionalhook: bool
+    tryfirst: bool
+    trylast: bool
+    specname: str | None
+
+class HookspecMarker:
+    project_name: Final[str]
+    def __init__(self, project_name: str) -> None: ...
+    @overload
+    def __call__(self, function: _F, firstresult: bool = ..., historic: bool = ..., warn_on_impl: Warning | None = ...) -> _F: ...
+    @overload
+    def __call__(
+        self, function: None = ..., firstresult: bool = ..., historic: bool = ..., warn_on_impl: Warning | None = ...
```

Comment:
Our policy has changed recently to prefer including default values wherever possible (it's not necessary for typecheckers, but it makes for a better experience for users of IDEs). [`stubdefaulter`](https://github.com/JelleZijlstra/stubdefaulter) can add most them automatically for you, so I won't bother commenting on each signature that's missing default values :-)